$NetBSD: patch-Makefile,v 1.5 2025/10/09 15:46:43 dkazankov Exp $

Make it possible to set install parameters
Fix install parameters
Fix ENABLE_SHARED position

--- Makefile.orig	2025-02-25 14:35:21.000000000 +0200
+++ Makefile
@@ -34,9 +34,6 @@
 SOURCE_DIR    := $(shell dirname "$(MAKEFILE_LIST)")
 LIB_DIR       = lib/
 
-# Load current setup if any
--include makefile.setup
-
 # target options for cross-build
 ifeq ($(HOST),$(TARGET))
 GTARGET=
@@ -45,6 +42,13 @@
 GTARGET=--target=$(TARGET)
 endif
 
+ENABLE_SHARED := $(shell gprbuild $(GTARGET) -c -q -p \
+	-P$(MAKEPREFIX)config/test_shared 2>/dev/null && echo "yes")
+
+# Load current setup if any
+-include makefile.setup
+
+GPRINSTALL_OPTIONS=
 INSTALLER=$(LIB_INSTALLER)
 EXEC_INSTALLER=$(INSTALLER) -XBUILD=${BUILD}
 
@@ -61,9 +65,6 @@
 MAKEPREFIX=$(SOURCE_DIR)/
 endif
 
-ENABLE_SHARED := $(shell gprbuild $(GTARGET) -c -q -p \
-	-P$(MAKEPREFIX)config/test_shared 2>/dev/null && echo "yes")
-
 ifeq ($(ENABLE_SHARED), yes)
    LIBGPR_TYPES=static shared static-pic
 else
@@ -72,13 +73,13 @@
 
 # Make sure Windows's "OS" environment variable does not cause
 # confusion for cross-Linux builds.
-LIBGPR_OS = $(if $(findstring linux,$(TARGET)),-XOS=UNIX)
+LIBGPR_OS = $(if $(or $(findstring linux,$(TARGET)),$(findstring bsd,$(TARGET))),-XOS=UNIX)
 
 # Used to pass extra options to GPRBUILD, like -d for instance
 GPRBUILD_OPTIONS=
 
 BUILDER=gprbuild -p -m $(GTARGET) $(RBD) -j${PROCESSORS} -XBUILD=${BUILD} ${GPRBUILD_OPTIONS}
-LIB_INSTALLER=gprinstall -p -f --target=$(TARGET) $(RBD) "--prefix=${prefix}"
+LIB_INSTALLER=gprinstall -p -f $(GTARGET) $(RBD) "--prefix=${prefix}" ${GPRINSTALL_OPTIONS}
 CLEANER=gprclean -q $(RBD)
 
 GPRBUILD_BUILDER=$(BUILDER) $(GPRBUILD_GPR) \
@@ -86,7 +87,7 @@
 LIBGPR_BUILDER=$(BUILDER) $(GPR_GPR) $(LIBGPR_OS)
 LIBGPR_INSTALLER=$(LIB_INSTALLER) $(GPR_GPR) $(LIBGPR_OS) -XBUILD=${BUILD} \
 	--install-name=gpr \
-	--build-var=LIBRARY_TYPE --build-var=GPR_BUILD $(GTARGET)
+	--build-var=LIBRARY_TYPE --build-var=GPR_BUILD
 LIBGPR_UNINSTALLER=$(LIB_INSTALLER) $(GPR_GPR) $(LIBGPR_OS) --install-name=gpr --uninstall
 
 #########
@@ -127,7 +128,7 @@
 install:
 	$(EXEC_INSTALLER) --mode=usage --install-name=gprbuild \
 		-XINSTALL_MODE=nointernal $(GPRBUILD_GPR)
-	$(EXEC_INSTALLER) --target=$(TARGET) --mode=usage  --install-name=gprbuild \
+	$(EXEC_INSTALLER) --mode=usage  --install-name=gprbuild \
 		-XINSTALL_MODE=internal $(GPRBUILD_GPR)
 
 complete: all install libgpr.install.static
